### ISSUE TITLE
Missing volume control in trailer preview

### ISSUE THREAD

**Issue by EinNiki** [stpnwf/ZestyTheme#10]
Fehlende Lautstärkeregelung in der Trailer-Vorschau

**Comment by stpnwf** [comment]
Can you attach a screenshot of the issue?
Does the issue only happen when watching trailers?
Are you on mobile or desktop?
On mobile, it is Jellyfin that doesn't include any volume control, Zesty does nothing there. On desktop I don't have any issues with it. Again, please share a screenshot or more details about the issue.

![Screenshot from 2024-10-20 22-03-49](https://github.com/user-attachments/assets/04c1ca78-3303-4f94-8b08-1066d909538a)

**Comment by EinNiki** [comment]
The problem only occurs on web/desktop as there is no way to adjust the volume there. I can only mute playback, but cannot adjust the volume.

**Comment by stpnwf** [comment]
> The problem only occurs on web/desktop as there is no way to adjust the volume there. I can only mute playback, but cannot adjust the volume.

Yeah, that is by design sadly. I would personally like to have the volume bar be displayed when you click on the volume button, but Jellyfin doesn't work like that. Clicking on the bar mutes the audio, since the bar is always visible.

Add this code. You may have to tweak the top: -4px to fit properly on your screen.

```
 .osdVolumeSliderContainer {
	  display: initial;
  }

  .osdVolumeSlider.mdl-slider.mdl-js-slider.mdl-slider-hoverthumb {
    display: initial;
    position: absolute;
    top: -4px !important;
  }
```

**Comment by mihawk90** [comment]
Not having a volume control at all seems more problematic then it not looking optimal IMO :sweat_smile: 

> I would personally like to have the volume bar be displayed when you click on the volume button, but Jellyfin doesn't work like that.

What about showing it on hover then?
```css
button.buttonMute:hover + .osdVolumeSliderContainer {
    display: initial
```

That would bring it up when hovered, but of course would need some additional rules to position the slider and everything.

**Comment by stpnwf** [comment]
> What about showing it on hover then?

That is interesting. I will look into it. Wouldn't moving the mouse to adjust the volume make it disappear though?


> Not having a volume control at all seems more problematic then it not looking optimal IMO 😅

Lol, I never ever use it. I always use my laptop's volume. So it is a non issue for me.
The rationale was that on your phone you will just use the volume buttons and on desktop you will use your laptop's volume buttons. I guess if you are on a PC it could be annoying.

**Comment by mihawk90** [comment]
> > What about showing it on hover then?
> 
> That is interesting. I will look into it. Wouldn't moving the mouse to adjust the volume make it disappear though?

Yes that's the main thing I ran into when I tried this briefly. I didn't have much time to play around with it but it should be possible by utilising transparent borders or `::before`/`::after` pseudo-elements. I *think* they also trigger the `:hover` state on their "real" parent elements.

**Comment by stpnwf** [comment]
> Yes that's the main thing I ran into when I tried this briefly. I didn't have much time to play around with it but it should be possible by utilising transparent borders or `::before`/`::after` pseudo-elements. I _think_ they also trigger the `:hover` state on their "real" parent elements. I know I've done something similar in the past but I don't remember how.

ok, Grok got it. Just can't for the life of me add more padding to the box, specially to the top. Will play around with it a little more.

![Screenshot from 2025-01-13 15-07-00](https://github.com/user-attachments/assets/0e5972b7-48da-46c6-bd5b-b38287b20424)


```
  /* Ensure the volume slider container is properly rotated and positioned */
  .osdVolumeSliderContainer {
    -webkit-flex-grow: 1;
    flex-grow: 1;
    min-width: 3em; /* Vertical height before rotation */
    min-height: 4em; /* Width before rotation */
    width: 6.5em; /* Actual width after rotation */
    height: 2em; /* Actual height after rotation */
    opacity: 0;
    visibility: hidden;
    position: absolute;
    transition: opacity 0.3s ease 1s, visibility 0s linear 1s;
    pointer-events: none;
    transform-origin: top left;
    transform: rotate(-90deg);
    background-color: rgba(var(--dark), 0.8);
    border-radius: var(--rounding);
    backdrop-filter: blur(3px);
  }
  /* Increase the hover area around the mute button */
  .buttonMute {
    position: relative;
  }
  .buttonMute::before {
    content: '';
    position: absolute;
    top: -10px;
    left: -10px;
    right: -10px;
    bottom: -10px;
    z-index: 1;
  }
  /* Make the volume slider container visible and interactive */
  .buttonMute:hover + .osdVolumeSliderContainer,
  .osdVolumeSliderContainer:hover {
    opacity: 1;
    visibility: visible;
    pointer-events: auto; /* Keep this interactive */
    transition: opacity 0.3s ease, visibility 0s;
  }
  /* Keep the volume slider interactive even when not directly hovered */
  .osdVolumeSliderContainer {
    /* Add this to maintain interaction */
    pointer-events: auto !important;
  }
  /* Ensure elements under the volume slider don't interfere */
  .player *:not(.osdVolumeSliderContainer, .buttonMute) {
    pointer-events: none;
  }
  /* But when hovering over non-volume controls, allow interaction */
  .player:hover:not(.osdVolumeSliderContainer):not(.buttonMute) * {
    pointer-events: auto;
  }
  /* Prevent other elements from interfering when volume slider is active */
  .player .osdVolumeSliderContainer:hover ~ * {
    pointer-events: none;
  }
  /* rotate volume number indicator */
  .osdVolumeSliderContainer .sliderBubbleTrack {
    left: 0;
    margin: 4.1em 5em;
    position: absolute;
    right: 0;
    transform: rotate(90deg);
  }
  /* tweak volume slider */
  .osdVolumeSlider.mdl-slider.mdl-js-slider.mdl-slider-hoverthumb {
    display: initial;
    position: absolute;
    left: 4% !important;
    max-width: 92% !important;
  }
  /* theme volume OSD */
    .iconOsd, .subtitleSyncContainer {
    -webkit-backdrop-filter: blur(5px);
    backdrop-filter: blur(5px);
    background: rgba(var(--dark),.8);
    border-radius: var(--rounding);
    color: rgba(var(--accent));
  }

```

**Comment by mihawk90** [comment]
Looks good already though, thanks for looking into it even if you're not using it :)

**Comment by dcm2610** [comment]
> I guess if you are on a PC it could be annoying.

I can +1 this haha, you can still use the mouse wheel or Up/Down Arrows to adjust volume but a volume slider on mouse hover would be nice!

**Comment by dcm2610** [comment]
Works on Firefox for the most part but is quite jank, definitely a step in the right direction though, works flawlessly on JMP

**Comment by mihawk90** [comment]
Hm weird I can't get it to work on JMP... reloaded page, removed the custom CSS and readded it, but the hover is not showing up.

I'm imagining it's a cache issue because jsdelivr isn't doing proper cache busting but I can't get into dev tools in JMP.

It shows on Chromium/Vivaldi but it looks a bit jank:

![Image](https://github.com/user-attachments/assets/6b3ccf03-f7b1-48f0-9fa7-6023c62fa404)

**Comment by stpnwf** [comment]
> Works on Firefox for the most part but is quite jank, definitely a step in the right direction though, works flawlessly on JMP

yeah. I don't know what to do tbh. My experience is mixed in FF. I guess most people will be using a Chrome based browser.

**Comment by mihawk90** [comment]
That's unrelated to the theme anyway, the same happens on stock JF (Purple Haze skin).

It seems the slider input just stops receiving the mouse input, but that'd have to be reported to jellyfin-web.
